docs/design/storage.md

```diff
@@ -0,0 +1,211 @@
+* If the storage download policy is configured, the value in the file path of the operation is considered to be the access path of the storage backend, rather than the local file path or other backend paths.
+
+### API
+
+To achieve our design purposes, we extend the `Operation` API with `Storage`.
+
+```go
+// OperationSpec defines the desired state of Operation.
+type OperationSpec struct {
+  // Processor describes how to register a operation processor into kubediag.
+  Processor Processor `json:"processor"`
+  // Dependences is the list of all depended operations required to be precedently executed.
+  // +optional
+  Dependences []string `json:"dependences,omitempty"`
+  // Storage represents the type of storage for operation results.
+  // Operation output will not be managed by any storage driver if nil.
+  // +optional
+  Storage *Storage `json:"storage,omitempty"`
+}
+
+// Storage represents the type of storage for operation results.
+type Storage struct {
+  // Type specify which storage to save object.
+  // Only support minio now.
+  Provisioner string `json:"provisioner"`
+  // OperationUploadFilePathKey is the key corresponding to the file to be stored in the operation. The storage driver gets the local file path to upload according to the content of this key.
+  // The value of this parameter configured in the actual diagnosis must be a local file path.
+  // If configured, the upload and storage operation will be executed.
+  // +optional
+  OperationUploadFilePathKey string `json:"operationUploadFilePathKey"`
+  // OperationDownloadFilePathKey is the key of the file to be downloaded in operation, and the storage driver downloads the file to the local according to the content of this key.
+  // The value of this parameter configured in the actual diagnosis must be a file path in the backend storage.
+  // If configured, it will perform the operation of downloading the file.
+  // After the download is complete, the file path corresponding to this key in diagnosis will be changed from the remote path to the local path.
+  // +optional
+  OperationDownloadFilePathKey string `json:"operationDownloadFilePathKey"`
+}
+```
+
+The storage is designed for users to define how should KubeDiag store their data. It contains the following fields:
+
+* `spec.storage.provisioner`: Provisioner should be declared as `minio` in operation.
+* `spec.storage.operationUploadFilePathKey`: OperationUploadFilePathKey is the key corresponding to the file to be stored in the operation. The storage driver gets the local file path to upload according to the content of this key.
+* `spec.storage.operationDownloadFilePathKey`: OperationDownloadFilePathKey is the key of the file to be downloaded in operation, and the storage driver downloads the file to the local according to the content of this key.
+
+## Implementation
+
+We currently implement support for minio storage. The following are the implementation details.
+
+### KubeDiag Agent Configuration
+
```

Comment:
The address and authentication information of minio can be stored in etcd through service domain and secret